src/Microsoft.DocAsCode.MarkdownLite/JsonTokenTreeRenderer.cs

```diff
@@ -0,0 +1,282 @@
+        {
+            var childContent = StringBuffer.Empty;
+            foreach (var item in token.Content)
+            {
+                childContent += renderer.Render(item);
+            }
+            return this.Insert("Em", childContent, Type.NonLeaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownCodeInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("InLineCode", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, GfmDelInlineToken token, MarkdownInlineContext context)
+        {
+            var childContent = StringBuffer.Empty;
+            foreach (var item in token.Content)
+            {
+                childContent += renderer.Render(item);
+            }
+            return this.Insert("Del", childContent, Type.NonLeaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTagInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("Tag", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownBrInlineToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("Br", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTextToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("InLineText", this.GetSize(), Type.Leaf);
+        }
+
+        #endregion
+
+        #region Misc
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownIgnoreToken token, IMarkdownContext context)
+        {
+            return this.Insert("Ignore", this.GetSize(), Type.Leaf);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownRawToken token, IMarkdownContext context)
+        {
+            return this.Insert($"Raw(From{token.Rule.Name})", GetSize(), Type.Leaf);
+        }
+
+        #endregion
+
+        #region Private
+
+        private enum Type
+        {
+            Leaf,
+            NonLeaf
+        }
+
+        private StringBuffer Insert(StringBuffer name, StringBuffer tokenContent, Type tokenType)
+        {
+            StringBuffer result = "{\"name\":\"";
+            result += name;
+            result += "\"";
+            switch (tokenType)
+            {
+                case Type.NonLeaf:
+                {
+                    // If tokenContent is not empty ,should remove the last character(',')
+                    if (tokenContent.GetLength() > 0 && tokenContent.EndsWith(','))
+                    {
+                        // TODO: add a method 'remove' of StringBuffer
+                        string contentTemp = tokenContent;
+                        tokenContent = contentTemp.Remove(contentTemp.Length - 1);
```

Comment:
The stringBuffer doesn't have a 'Remove' function

---
In reply to: [75074866](https://github.com/dotnet/docfx/pull/608#discussion_r75074866) [](ancestors = 75074866)